opencsp/app/sofast/SofastGUI.py

```diff
@@ -627,38 +592,28 @@ def run_gray_levels_cal(self) -> None:
         # Check system class exists
         self._check_system_loaded()
 
-        # Get selected calibration object
-        idx = self.cal_options.index(self.var_cal_select.get())
-        cal_object = self.cal_objects[idx]
+        # Get selected calibration type
+        cal_type = SofastService.cal_options[self.var_cal_select.get()]
 
-        # Capture images
-        def _func_0():
+        # Let us know when each phase is starting/has finished
+        def on_calibrating():
             print('Calibrating...')
-            self.system.run_display_camera_response_calibration(res=10, run_next=self.system.run_next_in_queue)
 
-        # Process data
-        def _func_1():
+        def on_processing():
             print('Processing calibration data')
-            # Get calibration images from System
-            calibration_images = self.system.get_calibration_images()[0]  # only calibrating one camera
-            # Load calibration object
-            self.calibration = cal_object.from_data(calibration_images, self.system.calibration_display_values)
-            # Save calibration object
-            self.calibration.save_to_hdf(file)
-            # Save calibration raw data
-            data = [self.system.calibration_display_values, calibration_images]
-            datasets = ['CalibrationRawData/display_values', 'CalibrationRawData/images']
-            hdf5_tools.save_hdf5_datasets(data, datasets, file)
-            # Show crosshairs
-            self.show_crosshairs()
-            # Display message
+
+        def on_processed():
             self.var_gray_lvl_cal_status.set('Calibration data: Loaded/Saved')
             print(f'Calibration complete. Results loaded and saved to\n    {file}')
-            # Continue
-            self.system.run_next_in_queue()
 
-        self.system.set_queue([_func_0, _func_1])
-        self.system.run_next_in_queue()
+        # Run calibration
+        self.service.run_gray_levels_cal(
+            calibration_class=cal_type,
+            calibration_hdf5_path_name_ext=file,
+            on_calibrating=on_calibrating,
```

Comment:
I believe the input "on_calibrating" should be "on_captured," (or the other way around). It doesn't look like SofastService doesn't have an "on_calibrating" input. 
```suggestion
            on_captured=on_calibrating,
```

Reply:
I don't know how you even saw this. Nice catch! Thanks!